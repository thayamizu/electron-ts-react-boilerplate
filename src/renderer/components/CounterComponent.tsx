import * as React from "react";
import { Button, Grid, Typography } from "@material-ui/core";

/**
 *
 *
 * @interface CounterProps
 */
export interface CounterProps {
    counter: number;
    onClickIncrement: (event: React.MouseEvent<HTMLElement>) => void;
    onClickDecrement: (event: React.MouseEvent<HTMLElement>) => void;
}

/**
 * Counter Component.
 */
export const Counter = ({
    counter,
    onClickIncrement,
    onClickDecrement,
}: CounterProps) => {
    return (
        <Grid>
            {/** カウンタ表示領域 */}
            <Grid container>
                <Grid item xs>
                    <Typography variant="h4" align="center">
                        {counter}
                    </Typography>
                </Grid>
            </Grid>

            {/** カウンタ操作領域 */}
            <Grid container>
                <Grid item xs style={{ padding: 8 }}>
                    <Button
                        fullWidth
                        onClick={onClickIncrement}
                        variant={`contained`}
                        color={"primary"}
                    >
                        increment
                    </Button>
                </Grid>
                <Grid item xs style={{ padding: 8 }}>
                    <Button
                        fullWidth
                        onClick={onClickDecrement}
                        variant={`contained`}
                        color={"secondary"}
                    >
                        decrement
                    </Button>
                </Grid>
            </Grid>
        </Grid>
    );
};
