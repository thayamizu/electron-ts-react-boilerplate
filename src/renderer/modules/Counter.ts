import * as ReduxActions from "redux-actions";

// Counter State Type
export interface CounterState {
    counter: number;
    index: n;
}

/**
 *
 *
 * @export
 * @class CounterModel
 */
export class CounterModel {
    // State of Counter
    private state: CounterState;

    /**
     *Creates an instance of CounterModel.
     * @memberof CounterModel
     */
    public constructor(state: CounterState) {
        this.state = state;
    }

    /**
     * Increment counter
     *
     * @returns {CounterModel}
     * @memberof CounterModel
     */
    public increment(): CounterModel {
        this.state.counter++;
        return new CounterModel(this.state);
    }

    /**
     * Decrement counter
     *
     * @returns {CounterModel}
     * @memberof CounterModel
     */
    public decrement(): CounterModel {
        this.state.counter--;
        return new CounterModel(this.state);
    }

    /**
     * Get counter state.
     *
     * @returns {CounterState}
     * @memberof CounterModel
     */
    public getState(): CounterState {
        return { ...this.state };
    }
}

/**
 * Counter Action Type Name
 *
 * @export
 * @enum {number}
 */
export enum CounterActionNames {
    Increment = "Counter/Increment",
    Decrement = "Counter/Decrement"
}

// Increment Action
export const increment = ReduxActions.createAction<void>(
    CounterActionNames.Increment
);

// Decrement Action
export const decrement = ReduxActions.createAction<void>(
    CounterActionNames.Decrement
);

export const counterReducer = ReduxActions.handleActions<CounterModel, void>(
    {
        [CounterActionNames.Increment]: (state: CounterModel): CounterModel => {
            return state.increment();
        },

        [CounterActionNames.Decrement]: (state: CounterModel): CounterModel => {
            return state.decrement();
        }
    },
    new CounterModel({ counter: 0 })
);
