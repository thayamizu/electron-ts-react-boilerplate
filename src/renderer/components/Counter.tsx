import * as React from "react";
import { Button, Grid } from "@material-ui/core";
import { CounterState } from "renderer/modules/Counter";
import { CounterActionDispatcher } from "../dispatcher/Counter";

/**
 *
 *
 * @interface CounterProps
 */
export interface CounterProps {
    state: CounterState;
    actions: CounterActionDispatcher;
}

/**
 * Counter Component.
 */
export class Counter extends React.Component<CounterProps, {}> {
    public constructor(props: CounterProps) {
        super(props);
    }

    /**
     * render
     */
    public render(): JSX.Element {
        return (
            <Grid
                container
                spacing={0}
                alignItems={"center"}
                justify={"center"}
                style={{ width: "100%" }}
            >
                <Grid item xs={12} style={{ textAlign: "center" }}>
                    <h1>Hello React World! </h1>
                    <h2>{this.props.state.counter}</h2>
                    <Button
                        onClick={this.props.actions.onClickIncrement}
                        variant={`contained`}
                        color={"primary"}
                    >
                        increment
                    </Button>
                    &emsp;
                    <Button
                        onClick={this.props.actions.onClickDecrement}
                        variant={`contained`}
                        color={"secondary"}
                    >
                        decrement
                    </Button>
                </Grid>
            </Grid>
        );
    }
}
