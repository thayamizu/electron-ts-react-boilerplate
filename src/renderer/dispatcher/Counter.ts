import * as ReduxActions from "redux-actions";
import * as Actions from "renderer/modules/Counter";

/**
 *
 *
 * @export
 * @class CounterActionDispatcher
 */
export class CounterActionDispatcher {
    private dispatch: (action: ReduxActions.Action<void>) => void;

    /**
     *Creates an instance of CounterController.
     * @param {Redux.Dispatch<Redux.Action<void>>} dispatch
     * @memberof CounterController
     */
    public constructor(dispatch: (action: ReduxActions.Action<void>) => void) {
        this.dispatch = dispatch;
    }

    /**
     * On click decrment button.
     *
     */
    public onClickDecrement = (event: React.MouseEvent<HTMLElement>): void => {
        this.dispatch(Actions.decrement());
    };

    /**
     * On click increment button.
     *
     */
    public onClickIncrement = (event: React.MouseEvent<HTMLElement>): void => {
        this.dispatch(Actions.increment());
    };
}
