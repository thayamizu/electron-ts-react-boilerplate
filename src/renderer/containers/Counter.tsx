import * as ReactRedux from "react-redux";
import * as Redux from "redux";
import * as Store from "renderer/Store";
import { Counter } from "renderer/components/Counter";
import { CounterActionDispatcher } from "renderer/dispatcher/Counter";

function mapStateToProps(state: Store.StoreType) {
    return { state: state.getState() };
}

function mapDispatchToProps(dispatch: Redux.Dispatch<Redux.Action>) {
    return { actions: new CounterActionDispatcher(dispatch) };
}

export default ReactRedux.connect(mapStateToProps, mapDispatchToProps)(Counter);
