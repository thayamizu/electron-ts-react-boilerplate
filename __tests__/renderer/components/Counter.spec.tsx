import * as Enzyme from "enzyme";
import * as React from "react";
import { Counter } from "renderer/components/Counter";
import { CounterActionDispatcher } from "renderer/dispatcher/Counter";

const Adapter = require("enzyme-adapter-react-16");
Enzyme.configure({ adapter: new Adapter() });

test("Counterコンポーネント", () => {
    const component = Enzyme.shallow(
        <Counter
            state={{ counter: 10 }}
            actions={new CounterActionDispatcher(jest.fn())}
        />
    );

    expect(component).toMatchSnapshot();
});
