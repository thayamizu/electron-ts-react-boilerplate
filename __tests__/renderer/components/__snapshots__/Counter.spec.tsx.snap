// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counterコンポーネント 1`] = `ShallowWrapper {}`;
